compiler/lib/global_deadcode.ml

```diff
@@ -352,7 +352,14 @@ let zero prog sentinal live_table =
     let branch =
       (* Zero out return values in last instruction, otherwise do nothing. *)
       match block.branch with
-      | Return x, loc -> Return (zero_var x), loc
+      | Return x, loc ->
```

Comment:
Maybe the comment should be updated as well to reflect the tail call special case?